src/core/Svg.tsx

```diff
@@ -0,0 +1,79 @@
+import { Object3DProps, useLoader } from '@react-three/fiber'
+import * as React from 'react'
+import { forwardRef, Fragment, useEffect, useMemo } from 'react'
+import { DoubleSide, Object3D } from 'three'
+import { SVGLoader } from 'three-stdlib'
+
+export interface SvgProps extends Object3DProps {
+  /** src can be a URL or SVG data */
+  src: string
+  skipFill?: boolean
+  skipStrokes?: boolean
+  strokesWireframe?: boolean
+  fillWireframe?: boolean
+}
+
+export const Svg = forwardRef<Object3D, SvgProps>(function R3FSvg(
+  { src, skipFill, skipStrokes, strokesWireframe, fillWireframe, ...props },
+  ref
+) {
+  const svg = useLoader(SVGLoader, !src.startsWith('<svg') ? src : `data:image/svg+xml;utf8,${src}`)
+
+  const strokeGeometries = useMemo(
+    () =>
+      skipStrokes
+        ? []
+        : svg.paths.map((path) =>
+            path.userData?.style.stroke === undefined || path.userData.style.stroke === 'none'
+              ? null
+              : path.subPaths.map((subPath) => SVGLoader.pointsToStroke(subPath.getPoints(), path.userData!.style))
+          ),
+    [svg, skipStrokes]
+  )
+
+  useEffect(() => {
+    return () => strokeGeometries.forEach((group) => group && group.map((g) => g.dispose()))
+  }, [strokeGeometries])
+
+  return (
+    <object3D ref={ref} {...props}>
+      <object3D scale={[1, -1, 1]}>
+        {svg.paths.map((path, p) => (
```

Comment:
you should not be using the `index` as the key for a map. Does the path not have a UUID?

Reply:
No, `SVGResultPaths` does not have a UUID, I tried. But in all fairness, is quite unlikely that the Svg will change in a way that could make `keys` useful in that way? 

On the other hand, I updated the Svg props to receive Material props so you can have a bit more control over them instead of exposing just the wireframe prop.

Reply:
`SVGResultPaths` extends `ShapePath` which has `currentPath` could that be used as the `key`?

Reply:
@joshuaellis `currentPath` is an object. No type definition so is hard to tell, but I did test it and I believe what it is returning is a [Path](https://threejs.org/docs/?q=path#api/en/extras/core/Path) instance. [Here is](https://github.com/mrdoob/three.js/blob/dev/src/extras/core/ShapePath.js#L21) when is initialised.

Reply:
ah, dang. Well thanks for doing all that :)